### DISCUSSION TITLE
No such file or directory: 'C:\\Program Files\\Meson\\mesonbuild\\scripts\\python_info.py'

### DISCUSSION THREAD

**Discussion by KyrietS** [mesonbuild/meson#11688] [Q&A] [answered]
Hi,

I tried compiling Cairo but when I ran `meson setup build/` I got an error:

```
meson.build:111:27: ERROR: <PythonExternalProgram 'python3' -> ['C:\\Program Files\\Meson\\meson.exe', 'runpython']> is not a valid python or it is missing distutils
```

Inside `build/meson-logs/meson-log.txt` I can see:

```
Program stderr:

Traceback (most recent call last):
  File "mesonbuild\mesonmain.py", line 194, in run
  File "mesonbuild\mesonmain.py", line 149, in run_runpython_command
  File "runpy.py", line 288, in run_path
  File "runpy.py", line 252, in _get_code_from_file
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Program Files\\Meson\\mesonbuild\\scripts\\python_info.py'
```

My specs:
* Windows 11
* Python 3.11.3
* Meson 1.1.0

I saw:

* https://github.com/mesonbuild/meson/issues/7702
* https://github.com/mesonbuild/meson/pull/11133

Did I do something wrong or just Meson is broken on the latest Python version?

**Comment by eli-schwartz** [comment]
The error here claims it cannot find the introspection script (which uses both sysconfig and distutils, so it's not related to the distutils deprecation).

The introspection script internally uses https://docs.python.org/3/library/importlib.resources.html to locate data, which in this case is supposed to return either the original file packed into the application or else a temporary file containing the contents, whichever is available. Seemingly, it is returning a file that doesn't actually exist. I'm not sure why...

**Reply by eli-schwartz** [reply, marked as answer]
OK, this is a bug in our collection hook. PyInstaller requires it to be listed as a data file.

**Reply by KyrietS** [reply]
I don't know how long it will take to produce a new installer or (maybe) bump a version of Meson, so here I give a workaround for other users:

```
cd your_project
git clone git@github.com:mesonbuild/meson.git
python -m meson.meson setup builddir/
``` 

It worked for me 👍

**Reply by eli-schwartz** [reply]
I would just use `pip install meson` for that. :)

There's also a meson.py script at the root of the git clone, which you can use as `python path/to/git/clone/meson.py setup builddir/`.

**Comment by eli-schwartz** [comment]
Copied to a bug report in #11689

**Reply by KyrietS** [reply]
Thank you for such a quick response! I will test your fix as soon as a new installer comes out 🤗
